libsql/src/hrana/hyper.rs

```diff
@@ -37,23 +37,19 @@ impl HttpSender {
             .header(AUTHORIZATION, auth)
             .header("x-libsql-client-version", self.version.clone())
             .body(hyper::Body::from(body))
-            .unwrap();
+            .map_err(|err| HranaError::Http(format!("{:?}", err)))?;
 
-        let res = self.inner.request(req).await.map_err(HranaError::from)?;
+        let res = self.inner.request(req).await?;
 
         if res.status() != StatusCode::OK {
-            let body = hyper::body::to_bytes(res.into_body())
-                .await
-                .map_err(HranaError::from)?;
-            let body = String::from_utf8(body.into()).unwrap();
-            return Err(HranaError::Api(body));
+            let body = hyper::body::to_bytes(res.into_body()).await?;
+            let msg = String::from_utf8(body.into()).unwrap_or_else(|err| err.to_string());
```

Comment:
maybe a bit more context could be added here to explain that the string error comes from an invalid payload

Reply:
After change: HranaError::Api(`Invalid payload: invalid utf-8 sequence of 1 bytes from index 1`)